packages/libs/react-ui/src/components/Grid/GridItem.tsx

```diff
@@ -1,40 +1,54 @@
-import { gridItemClass } from './Grid.css';
+import {
+  explicitItemColumnVariant,
+  gridItemClass,
+  itemColumnVariants,
+  ResponsiveInputType,
+  rowSpanVariants,
+} from './Grid.css';
 
-import React, { CSSProperties, FC, ReactNode } from 'react';
-
-export type ColSpanType = number | 'auto';
+import classNames from 'classnames';
+import React, { FC, ReactNode } from 'react';
 
 export interface IGridItemProps {
   children?: ReactNode;
-  colStart?: ColSpanType;
-  colEnd?: ColSpanType;
-  rowStart?: ColSpanType;
-  rowEnd?: ColSpanType;
-  area?: CSSProperties['gridArea'];
+  columnSpan?: ResponsiveInputType;
+  rowSpan?: keyof typeof rowSpanVariants;
 }
 
 const GridItem: FC<IGridItemProps> = ({
   children,
-  colStart,
-  colEnd,
-  rowStart,
-  rowEnd,
-  area,
+  columnSpan,
+  rowSpan = 1,
 }) => {
-  const styles =
-    area === undefined
-      ? {
-          gridColumnStart: colStart,
-          gridColumnEnd: colEnd,
-          gridRowStart: rowStart,
-          gridRowEnd: rowEnd,
-        }
-      : {
-          gridArea: area,
-        };
-
+  const className = classNames(
+    gridItemClass,
+    rowSpanVariants[rowSpan],
+    columnSpan &&
+      typeof columnSpan === 'number' &&
+      explicitItemColumnVariant[columnSpan],
+    columnSpan &&
+      typeof columnSpan !== 'number' &&
+      columnSpan.sm &&
+      itemColumnVariants.sm[columnSpan.sm],
+    columnSpan &&
+      typeof columnSpan !== 'number' &&
+      columnSpan.md &&
+      itemColumnVariants.md[columnSpan.md],
+    columnSpan &&
+      typeof columnSpan !== 'number' &&
+      columnSpan.lg &&
+      itemColumnVariants.lg[columnSpan.lg],
+    columnSpan &&
+      typeof columnSpan !== 'number' &&
+      columnSpan.xl &&
+      itemColumnVariants.xl[columnSpan.xl],
+    columnSpan &&
+      typeof columnSpan !== 'number' &&
+      columnSpan.xxl &&
+      itemColumnVariants.xxl[columnSpan.xxl],
+  );
```

Comment:
Is there a way to make this a bit more readable and easy to maintain?

Reply:
Yeah, I think there should be some way to make this easier to read and maintain since it is quite repetitive. Maybe pull it into a function?

Reply:
Moved both for `Grid.Root` and `Grid.Item` into a function:
https://github.com/kadena-community/kadena.js/blob/ceab74037d6140646af0f052e8519f0a12aec671/packages/libs/react-ui/src/components/Grid/GridItem.tsx#L18